velox/exec/tests/ExpressionBuilderTest.cpp

```diff
@@ -0,0 +1,193 @@
+  ExprWrapper result;
+
+  result = "a"_f == 10L;
+  EXPECT_EQ(result, parseSql("a = 10"));
+
+  result = "a"_f != 10L;
+  EXPECT_EQ(result, parseSql("a != 10"));
+
+  result = "a"_f < 10L;
+  EXPECT_EQ(result, parseSql("a < 10"));
+
+  result = "a"_f <= 10L;
+  EXPECT_EQ(result, parseSql("a <= 10"));
+
+  result = "a"_f > 10L;
+  EXPECT_EQ(result, parseSql("a > 10"));
+
+  result = "a"_f >= 10L;
+  EXPECT_EQ(result, parseSql("a >= 10"));
+
+  result = isNull("a"_f);
+  EXPECT_EQ(result, parseSql("a is null"));
+
+  result = isNull("a"); // this is "a" literal.
+  EXPECT_EQ(result, parseSql("\'a\' is null"));
+
+  result = !isNull("a"_f);
+  EXPECT_EQ(result, parseSql("a is not null"));
+
+  result = between("a"_f, 0L, 10L);
+  EXPECT_EQ(result, parseSql("a between 0 and 10"));
+
+  // Reverse order. As long as one side of operators are ExprWrapper, it
+  // generates expressions as expected.
+  result = 10L < "a"_f;
+  EXPECT_EQ(result, parseSql("10 < a"));
+}
+
+TEST(ExpressionBuilderTest, arithmetics) {
+  ExprWrapper result;
+
+  result = "b"_f + 1L;
+  EXPECT_EQ(result, parseSql("b + 1"));
+
+  result = "b"_f - 1L;
+  EXPECT_EQ(result, parseSql("b - 1"));
+
+  result = "b"_f * 1L;
+  EXPECT_EQ(result, parseSql("b * 1"));
+
+  result = "b"_f / 1L;
+  EXPECT_EQ(result, parseSql("b / 1"));
+
+  result = "b"_f % 1L;
+  EXPECT_EQ(result, parseSql("b % 1"));
+
+  result = "b"_f + 1L - 10L;
+  EXPECT_EQ(result, parseSql("b + 1 - 10"));
+
+  result = "b"_f * 1L / 10L;
+  EXPECT_EQ(result, parseSql("b * 1 / 10"));
+}
+
+TEST(ExpressionBuilderTest, conjuncts) {
+  ExprWrapper result;
+
+  result = "b"_f && 1L;
+  EXPECT_EQ(result, parseSql("b AND 1"));
+
+  result = "b"_f || 1L;
+  EXPECT_EQ(result, parseSql("b OR 1"));
+
+  result = "a"_f && "b"_f || "c"_f;
+  EXPECT_EQ(result, parseSql("a AND b OR c"));
+}
+
+TEST(ExpressionBuilderTest, functions) {
+  ExprWrapper result;
+
+  result = call("func");
+  EXPECT_EQ(result, parseSql("func()"));
+
+  result = call("func", field("a"));
+  EXPECT_EQ(result, parseSql("func(a)"));
+
+  result = call("func", field("a"), field("b"), field("c"));
+  EXPECT_EQ(result, parseSql("func(a, b, c)"));
+
+  result = call("func", literal(10L), literal(10.23), field("c"));
+  EXPECT_EQ(result, parseSql("func(10, 10.23, c)"));
+
+  // Nested functions.
+  result = call("f1", call("f2", "a"_f > call("f3", "d"_f)));
+  EXPECT_EQ(result, parseSql("f1(f2(a > f3(d)))"));
+}
+
+TEST(ExpressionBuilderTest, alias) {
+  ExprWrapper result;
+
+  result = alias(field("col"), "foo");
+  EXPECT_EQ(result, parseSql("col as foo"));
+
+  result = alias("col"_f, "foo");
+  EXPECT_EQ(result, parseSql("col as foo"));
+
+  result = alias(literal(1L), "bar");
+  EXPECT_EQ(result, parseSql("1 as bar"));
+
+  result = alias(1L, "bar");
+  EXPECT_EQ(result, parseSql("1 as bar"));
+}
+
+TEST(ExpressionBuilderTest, combined) {
+  ExprWrapper result;
+
+  result = 10L * "c1"_f > call("func", 3.4, "g"_f / "h"_f, call("j"));
```

Comment:
This expression is not nearly as complex as expressions commonly found in production, yet, even that is notably less readable as compared to SQL.

I think it is useful to have this API, but, perhaps, not as a replacement for SQL.